lib/ansible/module_utils/vmware.py

```diff
@@ -1225,6 +1225,30 @@ def find_datacenter_by_name(self, datacenter_name):
         """
         return find_datacenter_by_name(self.content, datacenter_name=datacenter_name)
 
+    def is_datastore_valid(self, datastore_obj=None):
+        """
+        Check if datastore selected is valid or not
+        Args:
+            datastore_obj: datastore managed object
+
+        Returns: True if datastore is valid, False if not
+        """
+        if not datastore_obj \
+           or datastore_obj.summary.maintenanceMode != 'normal' \
+           or not datastore_obj.summary.accessible:
+            return False
+
+        is_readonly = False
+        for host in datastore_obj.host:
```

Comment:
The user may have several hosts and only one with the datastore mounted RO. In this case, the datastore will never be picked.

Could you add a not to make clear this is not the best approach.

Reply:
I removed the logic from this API since it was not making sense here.